lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -780,6 +776,31 @@ impl LayerModel {
         layer
     }
 
+    /// Create a virtual sublayer of this layer. The order of virtual layer creation determines
+    /// depth ordering of their content objects of the type specified in the virtual layers' type
+    /// parameter.
+    pub fn create_virtual_sublayer<S: Shape>(
+        self: &Rc<Self>,
+        _name: impl Into<String>,
+    ) -> VirtualLayer<S> {
+        let system_id = ShapeSystem::<S>::id();
+        let mut partitions = self.virtual_sublayer_buffer_partitions.borrow_mut();
+        let index = partitions.entry(system_id).or_default();
+        let id = VirtualLayerId { index: *index };
+        *index += 1;
+        VirtualLayer { layer: WeakLayer { model: self.downgrade() }, id, shape: default() }
+    }
```

Comment:
A few naming problems, that actually I think are important:
1. We are using the term "sub-layer" currently for something different. Layers can have sub-layers, and I think this naming is correct. E.g. we are using it for masks.
2. I do not understand the word "virtual" here. Your layers are not virtual. They are very similar to the layers existing before this PR, which are not virtual either. I don't think we should name them "virtual". Sorry for not providing you with a better name, I don't know what is the correct name here yet, but I feel that "virtual" is confusing. Maybe it should be named `LayerSymbolOrdering`?